tests/failpoints/cases/test_merge.rs

```diff
@@ -308,6 +308,79 @@ fn test_node_merge_catch_up_logs_leader_election() {
     must_get_equal(&cluster.get_engine(3), b"k11", b"v11");
 }
 
+// Test if merge is working properly if no need to catch up logs,
+// also there may be a propose of compact log after prepare merge is proposed.
+#[test]
+fn test_node_merge_catch_up_logs_no_need() {
+    let _guard = crate::setup();
+    let mut cluster = new_node_cluster(0, 3);
+    configure_for_merge(&mut cluster);
+    cluster.cfg.raft_store.raft_base_tick_interval = ReadableDuration::millis(10);
+    cluster.cfg.raft_store.raft_election_timeout_ticks = 25;
+    cluster.cfg.raft_store.raft_log_gc_threshold = 12;
+    cluster.cfg.raft_store.raft_log_gc_count_limit = 12;
+    cluster.cfg.raft_store.raft_log_gc_tick_interval = ReadableDuration::millis(100);
+    cluster.run();
+
+    cluster.must_put(b"k1", b"v1");
+    cluster.must_put(b"k3", b"v3");
+
+    let pd_client = Arc::clone(&cluster.pd_client);
+    let region = pd_client.get_region(b"k1").unwrap();
+    let peer_on_store1 = find_peer(&region, 1).unwrap().to_owned();
+    cluster.must_transfer_leader(region.get_id(), peer_on_store1);
+    cluster.must_split(&region, b"k2");
+    let left = pd_client.get_region(b"k1").unwrap();
+    let right = pd_client.get_region(b"k2").unwrap();
+
+    // put some keys to trigger compact raft log
+    for i in 2..20 {
+        cluster.must_put(format!("k1{}", i).as_bytes(), b"v");
+        must_get_none(&cluster.get_engine(3), b"k1{}");
```

Comment:
Wrong key?

Reply:
we don't care the value of these keys, so just set `v`

Reply:
I think this key `b"k1{}"` is wrong, or please remove the `{}`.

Reply:
Oh, by mistake.